compendium/DeclarativeServices/src/manager/ConfigurationNotifier.cpp

```diff
@@ -165,78 +172,14 @@ namespace cppmicroservices
                     return false;
                 }
             } // release listenersMapHandle lock
+
             for (auto & mgr : mgrs)
             {
-                CreateFactoryComponent(pid, mgr);
+                componentFactory->CreateFactoryComponent(pid, mgr, properties);
```

Comment:
the way that `CreateFactoryComponent` now handles exceptions means that this loop could break early and not run `CreateFactoryComponent` on all `mgr` in `mgrs`, which is a change in behavior from before. Its not clear to me how exceptions and failure are supposed to be handled. can you clarify what is supposed to happen in the case of an exception being thrown?

Reply:
Hmmm. I don't think it's a change in behavior. If one factory component creation failed, they all failed. If you don't think this is the right behavior I can put a try /catch here so that the other components will still be created if one fails.

Reply:
I don't see how creation of one factory component causing none of the other unrelated factory components to be created is the right behavior. am I missing something?